sdk/src/main/scala/com/horizen/account/websocket/WebSocketAccountServerEndpoint.scala

```diff
@@ -0,0 +1,308 @@
+    }
+  }
+
+  private def subscribe(session: Session, rpcRequest: RpcRequest): Unit = {
+    val rpcParams = rpcRequest.params
+    val subscribeMethod = rpcParams.get(0)
+    val subscriptionId = createSubscriptionId()
+
+    subscribeMethod.asText() match {
+
+      case NEW_HEADS_SUBSCRIPTION.method =>
+        WebSocketAccountServerEndpoint.addNewHeadsSubscription(Subscription(session, subscriptionId))
+        log.debug("New Subscription on newHeads "+session.getId)
+        WebSocketAccountServerEndpoint.send(new RpcResponseSuccess(rpcRequest.id, subscriptionId), session)
+
+      case NEW_PENDING_TRANSACTIONS_SUBSCRIPTION.method =>
+        WebSocketAccountServerEndpoint.addNewPendingTransactionsSubscription(Subscription(session, subscriptionId))
+        log.debug("New Subscription on newPendingTransactions "+session.getId)
+        WebSocketAccountServerEndpoint.send(new RpcResponseSuccess(rpcRequest.id, subscriptionId), session)
+
+      case LOGS_SUBSCRIPTION.method =>
+        if (rpcParams.size() < 2)
+          WebSocketAccountServerEndpoint.send(new RpcResponseError(rpcRequest.id,
+            new RpcError(RpcCode.InvalidParams, "Missing filters (address, topcis).", "")),
+            session)
+        val logs = rpcParams.get(1)
+        val addressFilter = getLogAddresses(logs)
+        val topicFilter = getLogTopics(logs)
+        WebSocketAccountServerEndpoint.addLogsSubscription(SubscriptionWithFilter(session, subscriptionId,
+          addressFilter,
+          topicFilter
+        ))
+        log.debug("New Subscription on logs "+session.getId)
+        WebSocketAccountServerEndpoint.send(new RpcResponseSuccess(rpcRequest.id, subscriptionId), session)
+
+      case uknownMethod =>
+        WebSocketAccountServerEndpoint.send(new RpcResponseError(rpcRequest.id,
+          new RpcError(RpcCode.InvalidParams, "unsupported subscription type " + uknownMethod, "")),
+          session)
+        log.debug("unsupported subscription type " + uknownMethod)
+    }
+
+  }
+
+  private def unsubscribe(session: Session, rpcRequest: RpcRequest): Unit = {
+    val rpcParams = rpcRequest.params
+
+    val subscriptionIdsToRemove: util.ArrayList[String] = new util.ArrayList[String]()
+    rpcParams.forEach(subscriptionId => subscriptionIdsToRemove.add(subscriptionId.asText()))
+    val removedSubscription = WebSocketAccountServerEndpoint.removeSubscriptions(subscriptionIdsToRemove.toArray(new Array[String](0)))
+    if (!removedSubscription) {
+      WebSocketAccountServerEndpoint.send(new RpcResponseError(rpcRequest.id,
+        new RpcError(RpcCode.InvalidParams, s"Subscription ID not found.", "")),
+        session)
+    } else
+      WebSocketAccountServerEndpoint.send(new RpcResponseSuccess(rpcRequest.id, true), session)
+  }
+
+  private def createSubscriptionId(): String = {
+    Numeric.toHexStringWithPrefix(BigInteger.valueOf(WebSocketAccountServerEndpoint.subscriptionCounter.incrementAndGet()))
+  }
```

Comment:
here we may return BigInteger and it will be mapped automatically by EthMapper.

Reply:
Done